pkg/component/apiserver/types.go

```diff
@@ -76,12 +76,24 @@ type AuditWebhook struct {
 	Version *string
 }
 
+// AutoscalingMode represents the different autoscaling modes for an API Server.
+type AutoscalingMode int8
+
+const (
+	// AutoscalingModeBaseline differs substantially between kube-apiserver and gardener-apiserver.
+	// For kube-apiserver, it is active HPA, plus VPA in passive, recommend-only mode.
+	// For gardener-apiserver, it is VPA only.
```

Comment:
Why don't we add this HPA for GAPI now as well so that this is harmonized? I understand that with GEP-23, this `gardener-custom-metrics` component was hard-coded for seeds, but this is a new approach now, so nothing is hindering us from doing it for GAPI (and the KAPI of the virtual garden cluster) right away, or? Conceptually, GAPI/KAPI are "the same" anyways.

Reply:
Yes, I know that I have to do it for the gardener-operator as well. Before you posted your comment I added the following card https://github.com/orgs/gardener/projects/6#card-92504380 and I am planning to work on it next.

If you ask me, I don't think we have to hold this PR (we hope that it will bring utilization improvements for the kube-apiservers) just because of the gardener-operator part (2 HVPAs - for virtual-kube-apiserver and gardener-apiserver).

But if there is a common agreement that it should be done in this PR, I can do it in this PR.

cc @vlerenc @voelzmo

Reply:
Please keep in mind, that we now do something that is against the official (Kubernetes docs) recommendation and we touch the auto-scaling that we haven't touched in many, many years. So, while I am very optimistic (the recommendation was very general and we do something very specific to avoid the downsides) and the potential/projected savings are very high, we like to do it very, very carefully, i.e. seed by seed, observe carefully utilization and availability. In that sense, any downsides that affect the GAPI that we didn't see coming, will effect the entire landscape, not just the KAPIs on a particular seed.

Other than that, I have no comments. If for practical reasons it makes sense to combine that or not is your call, but when we roll it out, we will switch the GAPI probably last and only after we completed our careful "canary-like" "staged" seed roll-out (successfully, of course :-)).

Reply:
In the light of the tracking project card, I am fine to do it in a separate PR.